examples/energy-management-app/energy-management-common/src/ElectricalPowerMeasurementDelegate.cpp

```diff
@@ -0,0 +1,442 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <ElectricalPowerMeasurementDelegate.h>
+#include <app/reporting/reporting.h>
+
+#include <app/clusters/electrical-power-measurement-server/electrical-power-measurement-server.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::DataModel;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ElectricalPowerMeasurement;
+
+CHIP_ERROR ElectricalPowerMeasurementInstance::Init()
+{
+    return Instance::Init();
+}
+
+void ElectricalPowerMeasurementInstance::Shutdown()
+{
+    Instance::Shutdown();
+}
+
+// --------------- Internal Attribute Set APIs
+CHIP_ERROR ElectricalPowerMeasurementDelegate::SetPowerMode(PowerModeEnum newValue)
+{
+    PowerModeEnum oldValue = mPowerMode;
+
+    if (newValue >= PowerModeEnum::kUnknownEnumValue)
```

Comment:
There can be valid values larger than kUnknownEnumValue.

If newValue is coming from IM decoding, all unknown values have already been converted to kUnknownEnumValue and you can just test for equality.  If not, you should probably use
```
   if (EnsureKnownEnumValue(newValue) == PowerModeEnum::kUnknownEnumValue)
```